cmd/lakefs/cmd/run.go

```diff
@@ -398,6 +401,55 @@ func checkRepos(ctx context.Context, logger logging.Logger, authMetadataManager
 	}
 }
 
+func deleteExpiredAddressesJob(ctx context.Context, c *catalog.Catalog) {
+	s := gocron.NewScheduler(time.UTC)
+	_, err := s.Every(1).Hour().Do(func() {
+		deleteExpiredAddressTokens(ctx, c)
+	})
+	if err != nil {
+		logging.Default().WithError(err).Debug("failed execute delete expired addresses job")
+	}
+
+	s.StartBlocking()
+}
+
+func deleteExpiredAddressTokens(ctx context.Context, c *catalog.Catalog) {
+	const (
+		jobWorkers = 3
+		size       = 10
+	)
+	rChan := make(chan *catalog.Repository, size)
+	for i := 0; i < jobWorkers; i++ {
+		go rWorker(ctx, c, rChan)
+	}
+
+	hasMore := true
+	next := ""
+
+	for hasMore {
+		var err error
+		var repos []*catalog.Repository
+		repos, hasMore, err = c.ListRepositories(ctx, -1, "", next)
+		if err != nil {
+			logging.Default().WithError(err).Debug("failed to list repositories while deleting expired tokens")
+		}
+
+		for _, repo := range repos {
+			rChan <- repo
+			next = repo.Name
+		}
+	}
+}
+
+func rWorker(ctx context.Context, c *catalog.Catalog, rChan <-chan *catalog.Repository) {
+	for r := range rChan {
+		err := c.DeleteExpiredAddressTokens(ctx, r.Name)
```

Comment:
I think this function needs to be as part of this PR, as far as I understand only this flow should invoke DeleteExpiredAddressTokens

Reply:
done